### PR TITLE
Add Post Function for Comments

### PR DESCRIPTION
**Pull request by lornaarmstrong**
Task 4 from the Week 3 work.

### CODE REVIEW

**Comment by lornaarmstrong** [comment]
here's a screenshot showing the changes:

Screenshot 2020-07-23 at 6 03 14 PM - Display 1

File: portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -227,3 +226,34 @@ footer {
...
+}
+
+#commentText {
+  padding-bottom: 10px;
+}
+
+.commentSection {
+  margin-bottom: 75px;
+  width: 100%;
+}
```
Comment (MostafaGabriel) [inline portfolio/src/main/webapp/style.css:251]: nit: [separate words in ID and class names by a hyphen](https://google.github.io/styleguide/htmlcssguide.html#ID_and_Class_Name_Delimiters), they are usually in a lower case too. [Further reading](https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/)

File: portfolio/src/main/webapp/comment.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+import java.util.ArrayList;
+
+package com.google.sps.data;
```
Comment (MostafaGabriel) [inline portfolio/src/main/webapp/comment.java:4]: please move the file to `portfolio/src/main/java/com/google/sps/date/` then :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -18,30 +18,43 @@
...
+    response.sendRedirect("/index.html");
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
```
Comment (MostafaGabriel) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: 👏

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -18,30 +18,43 @@
...
     response.setContentType("application/json;");
+    String json = new Gson().toJson(messages);
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String fname = getParameter(request, "fname", "");
+    String lname = getParameter(request, "lname", "");
```
Comment (MostafaGabriel) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: nitty nit: we usually document ambiguous literal arguments at the call site just to know what a method will do. For example `getParameter(request, "fname", /* defaultValue= */ "")`.
Other examples of ambiguous literal arguments are `0`, `""`, `true` and `false`, or `null`

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,9 +12,22 @@
...
+
+function getComments(){
+  fetch('/data').then(response => response.json()).then((comments) => {
+    const commentContainer = document.getElementById("comment-container");
+    var commentHTML = "";
+    for (var i = 0; i < comments.length; i++){
+        commentHTML += "<div id = \"userComment\">";
+        commentHTML += "<h4 id = \"commentorName\">" + comments[i][0] + " " + comments[i][1] + "</h4>";
+        commentHTML += "<p id = \"commentText\">" + comments[i][2] + "</p>";
+        commentHTML += "</div>"
```
Comment (MostafaGabriel) [inline portfolio/src/main/webapp/script.js:29]: bonus: consider using [Node.appendChild()](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) here instead

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -18,30 +18,43 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Arrays;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+//import com.google.sps.data.Comment;
```
[the remark this answers is not in the document]
Comment (lornaarmstrong) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:26]: Will do, thanks!

File: portfolio/src/main/webapp/comment.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+public final class Comment {
```
[the remark this answers is not in the document]
Comment (lornaarmstrong) [reply portfolio/src/main/webapp/comment.java:6]: This was a section of code from the next step - I accidentally went too far and forgot to remove it for this code review.

File: portfolio/src/main/webapp/comment.java (hunk 2 above)
Comment (lornaarmstrong) [reply portfolio/src/main/webapp/comment.java:4]: Have done, thank you.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (lornaarmstrong) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: okay, thanks!

File: portfolio/src/main/webapp/style.css (hunk 1 above)
Comment (lornaarmstrong) [reply portfolio/src/main/webapp/style.css:251]: Okay, thanks Mostafa!

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -227,3 +226,34 @@ footer {
...
+.label {
+  margin-bottom: 10px;
+}
+
+#userComment {
+  background-color: rgb(240, 240, 240);
+  padding: 0px 15px;
+}
+
+commentorName {
```
[the remark this answers is not in the document]
Comment (lornaarmstrong) [reply portfolio/src/main/webapp/style.css:239]: okay, thank you!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (lornaarmstrong) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: Okay, thank you!

**Pull request merged**